operations/observability/Observability.md

```diff
@@ -0,0 +1,365 @@
+  "Description": "Validates that 'cray --version' is available and executes successfully on the local system.",
+  "Execution Time (nanoseconds)": 1312368478,
+  "Execution Time (seconds)": 1.312368478,
+  "Node": "ncn-m001",
+  "Result Code": 0,
+  "Result String": "PASS",
+  "Source": "http://ncn-m001.hmn:8997/ncn-healthcheck-master",
+  "Test Name": "Command 'cray --version' Available",
+  "Test Summary": "Command: command_available_1_cray_version: exit-status: matches expectation: [0]"
+}
+```
+
+### Grok-exporter deployment, service, and service-monitor
+
+Grok-exporter is deployed on all of the Kubernetes master nodes, by using node affinity in its deployment.
+The service for the grok-exporter is accessible at port `9144`. Service-monitor implementation is for Prometheus to access the metrics that are created by the grok-exporter.
+
+### Configuration file for the grok-exporter
+
+The configuration file for the grok-exporter parses the Goss test log files and creates metrics from them.
+grok-exporter version 3 configuration is used for this task and to match log expressions with regular expressions.
+
+The following is an example configuration passed to the grok-exporter to get metrics:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example_name
+  namespace: example_namespace
+data:
+  config.yml: |-
+    global:
+      config_version: 3
+    input:
+      type: file
+      paths: 
+      - /logs/goss_tests/*
+      fail_on_missing_logfile: false
+    grok_patterns:
+      - 'REGEX [regular].*expression'
+      - 'EXAMPLE [aA-zZ-].*'
+    metrics:
+      - type: gauge
+        name: example_metric
+        help: EXAMPLE METRIC HELP
+        match: '{"log_file": "%{REGEX:logfile}", "TEST": "%{EXAMPLE:ex}"}'
+        value: '{{`{{.logfile}}`}}'
+        labels:
+          example_test: '{{`{{.ex}}`}}'
+    server:
+     port: 9144
+```
+
+### Prometheus metrics and Grafana dashboard
+
+After the preceding steps the `goss_tests` metrics are seen in Prometheus when Goss tests are run. Using these metrics,
+Grafana dashboards are created to shows the Goss tests details visually.
+
+Goss test dashboard features:
+
+- The dashboard has a dropdown for products, suites, and tests.
+- By default, all products, suites, and tests are selected for overall Goss tests result.
+- The overall product result, total number of the products, products passed, products failed, and its execution time is seen.
+- Suite results and test results are seen from the dashboard.
+- Node-wise test results are seen with failed nodes, passed nodes, and the suite/test description.
+
+![Goss test dashboard](../../img/operations/GossTestsDashboard.png "Goss Test Dashboard")
+
+## Error dashboard
+
+Error dashboards provide key glance at-a-insights about what is broken and needs attention.
+
+The automatic generation of the desired Prometheus grok-exporter configuration is based upon passing in a set of regular expression patterns to detect.
```

Comment:
These grok-exporter configurations are automatically generated on the system? They aren't something included with CSM? Who is passing in these regular expressions on the system to generate the configurations?

Reply:
For generating grok-exporter configuration we need to pass set of regex in conig file. We have created regex pattern to detect.

Reply:
But this sentence as written is talking about automatic generation of the grok-exporter configuration. It sounds like you're saying that this generation is something you've already done, not something that actually happens on the live system. If that's the case, why is this sentence talking about automatically generating the configuration?

Reply:
changed